### PR TITLE
Fix flaky tests in RetryingHttpClientTest

### LINKED ISSUE #666: Flaky test: RetryingHttpClientTest
```
RetryingHttpClientTest.differentResponseTimeout: FAILURE
--------------------------------------------------------
com.linecorp.armeria.client.retry.RetryingHttpClientTest > differentResponseTimeout FAILED
    java.util.concurrent.CompletionException: com.linecorp.armeria.client.ResponseTimeoutException
        at java.util.concurrent.CompletableFuture.reportJoin(CompletableFuture.java:375)
        at java.util.concurrent.CompletableFuture.join(CompletableFuture.java:1934)
        at com.linecorp.armeria.client.retry.RetryingHttpClientTest.differentResponseTimeout(RetryingHttpClientTest.java:363)

        Caused by:
        com.linecorp.armeria.client.ResponseTimeoutException
```

```
RetryingHttpClientTest.consecutiveRequests: FAILURE
---------------------------------------------------
com.linecorp.armeria.client.retry.RetryingHttpClientTest > consecutiveRequests FAILED
    java.lang.AssertionError:
    Expecting:
     <499L>
    to be greater than or equal to:
     <500L>
        at com.linecorp.armeria.client.retry.RetryingHttpClientTest.consecutiveRequests(RetryingHttpClientTest.java:318)
```

```
RetryingHttpClientTest.timeoutWhenServerSendServiceUnavailable: FAILURE
-----------------------------------------------------------------------
com.linecorp.armeria.client.retry.RetryingHttpClientTest > timeoutWhenServerSendServiceUnavailable FAILED
    java.lang.AssertionError: 
    Expecting:
     <999L>
    to be greater than or equal to:
     <1000L> 
        at com.linecorp.armeria.client.retry.RetryingHttpClientTest.timeoutWhenServerSendServiceUnavailable(RetryingHttpClientTest.java:306)
```

### PR DESCRIPTION
**Pull request by minwoox**
Motivation:

Need to fix flaky tests.

Modifications:

- Relax the assertions a little bit, like in #636
- Increase the timing tolerance of responseTimeout and reduce backoff delay so
  when unexpected exception happens which adds 3200 milliseconds to test, still pass the test

Result:

- Fix #666

### CODE REVIEW

**Comment by trustin** [comment]
AppVeyor build failed due to unrelated flaky test: `ZooKeeperRegistrationTest.testServerNodeCreateAndDelete` (i.e. never mind.)

File: core/src/main/java/com/linecorp/armeria/client/retry/RetryingHttpClient.java [hunk 1]
```diff
@@ -169,13 +169,15 @@ private void retry0(int currentAttemptNo, Backoff backoff, ClientRequestContext
...
+            rootReqDuplicator.close();
             rootResponse.close(exception);
         } else {
             final EventLoop eventLoop = ctx.contextAwareEventLoop();
             nextDelay = Math.max(nextDelay, millisAfter);
             try {
                 nextDelay = getNextDelay(nextDelay, ctx);
             } catch (ResponseTimeoutException e) {
+                rootReqDuplicator.close();
                 rootResponse.close(e);
```
Comment (trustin) [inline core/src/main/java/com/linecorp/armeria/client/retry/RetryingHttpClient.java]: How about closing `rootResponse` before closing the duplicator? #667 will make `duplicator.close()` abort the root stream.

File: core/src/main/java/com/linecorp/armeria/client/retry/RetryingHttpClient.java [hunk 2]
```diff
@@ -169,13 +169,15 @@ private void retry0(int currentAttemptNo, Backoff backoff, ClientRequestContext
                         DefaultHttpResponse rootResponse, Throwable exception, long millisAfter) {
         long nextDelay = backoff.nextDelayMillis(currentAttemptNo);
         if (nextDelay < 0) {
+            rootReqDuplicator.close();
             rootResponse.close(exception);
```
Comment (trustin) [inline core/src/main/java/com/linecorp/armeria/client/retry/RetryingHttpClient.java]: How about closing `rootResponse` before closing the duplicator? #667 will make `duplicator.close()` abort the root stream.

Comment (minwoox) [inline core/src/main/java/com/linecorp/armeria/client/retry/RetryingHttpClient.java]: I think rootReqDuplicator.close() ahead of rootResponse.close() is better even thought it doesn't matter. The flow is like this: 1. client send HttpRequest 2. HttpRequest is wrapped by Duplicator 3. ChildRequest is sent to the server while rootResponse is returned to the client 4. If the RetryingClient doesn't decide to retry again ChildRequest is already subscribed so don't need the Duplicator 5. So clean the reqDuplicator before returning exception to the client..

**Comment by minwoox** [comment]
I will push the pr again so it do the AppVeyor build again.

File: core/src/main/java/com/linecorp/armeria/client/retry/RetryingHttpClient.java (hunk 2 above)
Comment (minwoox) [inline core/src/main/java/com/linecorp/armeria/client/retry/RetryingHttpClient.java]: I will change the order~ Thank you~

**Comment by trustin** [comment]
Thanks, @minwoox !

**Pull request merged**